### PR TITLE
IWCoG merge

### PR DESCRIPTION
**Pull request by asmith-ceng**
IWCoG algorithm is ready for review

### CODE REVIEW

**Comment by markasoftware** [comment]
I have been working on some substantial changes to the way the command line interface and stuff works, i will merge this after I'm done with that.

File: src/centroiders.cpp [hunk 1]
```diff
@@ -120,6 +118,127 @@ std::vector<Star> CenterOfGravityAlgorithm::Go(unsigned char *image, int imageWi
...
+                }
+            }
+            fwhm = sqrt((float) count);
+            standardDeviation = fwhm / (2.0 * sqrt(2.0 * log(2.0)));
+            float guessXCoord = (float) (p.guess % imageWidth);
+            float guessYCoord = (float) (p.guess / imageWidth);
+            //how much our new centroid estimate changes w each iteration
+            float change = 100.0;
+            //while we see some large enough change in estimated, maybe make it a global variable
+            while (change > 0.0001) {
```
Comment (markasoftware) [inline src/centroiders.cpp:213]: Maybe add it as a private member of the `IterativeWeightedCenterOfGravityAlgorithm` class, then prompt for it in `IWCoGCentroidAlgorithmPrompt`?

File: src/centroiders.cpp [hunk 2]
```diff
@@ -120,6 +118,127 @@ std::vector<Star> CenterOfGravityAlgorithm::Go(unsigned char *image, int imageWi
...
+                if (image[starIndices.at(j)] > p.maxIntensity / 2) {
+                    count++;
+                }
+            }
+            fwhm = sqrt((float) count);
+            standardDeviation = fwhm / (2.0 * sqrt(2.0 * log(2.0)));
+            float guessXCoord = (float) (p.guess % imageWidth);
+            float guessYCoord = (float) (p.guess / imageWidth);
+            //how much our new centroid estimate changes w each iteration
+            float change = 100.0;
```
Comment (markasoftware) [inline src/centroiders.cpp:211]: Prefer `INFINITY` for things like this.

File: src/centroiders.cpp [hunk 3]
```diff
@@ -120,6 +118,127 @@ std::vector<Star> CenterOfGravityAlgorithm::Go(unsigned char *image, int imageWi
...
+            //while we see some large enough change in estimated, maybe make it a global variable
+            while (change > 0.0001) {
+            //traverse through star indices, calculate W at each coordinate, add to final coordinate sums
+                for (int j = 0; j < (int) starIndices.size(); j++) {
+                    //calculate w
+                    float currXCoord = (float) (starIndices.at(j) % imageWidth);
+                    float currYCoord = (float) (starIndices.at(j) / imageWidth);
+                    float modifiedStdDev = 2.0 * pow(2, standardDeviation);
```
Comment (markasoftware) [inline src/centroiders.cpp:219]: the pow is backwards lol

File: src/centroiders.cpp [hunk 4]
```diff
@@ -120,6 +118,127 @@ std::vector<Star> CenterOfGravityAlgorithm::Go(unsigned char *image, int imageWi
...
+            float guessXCoord = (float) (p.guess % imageWidth);
+            float guessYCoord = (float) (p.guess / imageWidth);
+            //how much our new centroid estimate changes w each iteration
+            float change = 100.0;
+            //while we see some large enough change in estimated, maybe make it a global variable
+            while (change > 0.0001) {
+            //traverse through star indices, calculate W at each coordinate, add to final coordinate sums
+                for (int j = 0; j < (int) starIndices.size(); j++) {
+                    //calculate w
+                    float currXCoord = (float) (starIndices.at(j) % imageWidth);
```
Comment (markasoftware) [inline src/centroiders.cpp:217]: You don't have to fix it, but for future reference vectors support the indexing operator, `starIndices[j]`

File: src/centroiders.cpp [hunk 5]
```diff
@@ -120,6 +118,127 @@ std::vector<Star> CenterOfGravityAlgorithm::Go(unsigned char *image, int imageWi
...
+            int yDiameter = 0;
+            float yWeightedCoordMagSum = 0; 
+            float xWeightedCoordMagSum = 0; 
+            float weightedMagSum = 0; 
+            float fwhm; //fwhm variable
+            float standardDeviation;
+            float w; //weight value
+            
+            //computes indices to skip after done w current star
+            int step = i;
```
Comment (markasoftware) [inline src/centroiders.cpp:184]: What's the point of this? We already check `p.checkedIndices.count(i) == 0` on line 171?

File: src/centroiders.cpp [hunk 6]
```diff
@@ -76,17 +76,15 @@ int DetermineCutoff(unsigned char *image, int imageWidth, int imageHeight) {
 
 std::vector<Star> CenterOfGravityAlgorithm::Go(unsigned char *image, int imageWidth, int imageHeight) const {
     CentroidParams p;
-    std::unordered_set<int> checkedIndices;
+    //std::unordered_set<int> checkedIndices;
```
Comment (markasoftware) [inline src/centroiders.cpp:79]: Now that you've figured out this bug, let's remove the comments

Comment (asmith-ceng) [reply src/centroiders.cpp:79]: done

File: src/centroiders.cpp (hunk 3 above)
Comment (asmith-ceng) [reply src/centroiders.cpp:219]: done

File: src/centroiders.cpp (hunk 2 above)
Comment (asmith-ceng) [reply src/centroiders.cpp:211]: done

File: src/centroiders.cpp (hunk 5 above)
Comment (asmith-ceng) [reply src/centroiders.cpp:184]: done, also got rid of a simillar method in CoG

**Comment by markasoftware** [comment]
Are you planning on making any more changes or should I merge this?

**Comment by asmith-ceng** [comment]
might as well just merge it

**Pull request merged**